src/wallet/wallet.cpp

```diff
@@ -1349,24 +1394,34 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlock* pblock)
     MarkAffectedTransactionsDirty(tx);
 }
 
-void CWallet::MarkAffectedTransactionsDirty(const CTransaction& tx)
-{
+void CWallet::MarkAffectedTransactionsDirty(const CTransaction& tx) {
     // If a transaction changes 'conflicted' state, that changes the balance
     // available of the outputs it spends. So force those to be
     // recomputed, also:
-    BOOST_FOREACH(const CTxIn& txin, tx.vin)
+    BOOST_FOREACH(
+    const CTxIn &txin, tx.vin)
     {
         if (mapWallet.count(txin.prevout.hash))
             mapWallet[txin.prevout.hash].MarkDirty();
     }
-    for (const JSDescription& jsdesc : tx.vjoinsplit) {
-        for (const uint256& nullifier : jsdesc.nullifiers) {
+    for (const JSDescription &jsdesc : tx.vjoinsplit) {
+        for (const uint256 &nullifier : jsdesc.nullifiers) {
             if (mapSproutNullifiersToNotes.count(nullifier) &&
-                    mapWallet.count(mapSproutNullifiersToNotes[nullifier].hash)) {
+                mapWallet.count(mapSproutNullifiersToNotes[nullifier].hash)) {
                 mapWallet[mapSproutNullifiersToNotes[nullifier].hash].MarkDirty();
             }
         }
     }
+
+    // TODO: Do we want this?
+    for (const OutputDescription &output : tx.vShieldedOutput) {
+        // TODO: compute nullifier and check against...
+        uint256 nullifier; // TODO: compute...
```

Comment:
No need; iterate over `tx.vShieldedSpend`, and using the nullifier there. The intent of this function is to find transactions that generated coins or notes being spent by this one, and mark them as dirty.

Reply:
Yep, as discussed in meeting this will be updated.